relayer/shannon/mode_trusted.go

```diff
@@ -0,0 +1,148 @@
+package shannon
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/types/bech32"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sdk "github.com/pokt-network/shannon-sdk"
+)
+
+var (
+	// TrustedMode implements the OperationMode interface consumed by the Protocol struct.
+	// See the file mode.go for a detailed description of the Operation Mode concept.
+	_ OperationMode = &TrustedMode{}
+
+	// trustedModeInstance implements the OperationModeInstance interface consumed by the Protocol struct.
+	// It is built by the TrustedMode struct based on a specific request's details (HTTP headers as of now).
+	_ OperationModeInstance = &trustedModeInstance{}
+)
+
+const (
+	// headerAppAddress is the key of the entry in HTTP headers that holds the target app's address.
+	// The target app will be used to sign the relay request.
+	headerAppAddr = "app_address"
+)
+
+// NewTrustedMode initializes an instance of TrustedMode using the supplied list of private keys.
+// The supplied account client is required to instantiate relay request signers corresponding to the apps held by the Trusted operation mode.
+func NewTrustedMode(appsPrivateKeys []*secp256k1.PrivKey, accountClient sdk.AccountClient) (*TrustedMode, error) {
```

Comment:
`NewTrustedMode` should not have knowledge about the apps private keys.

Reply:
Thank you, fixed.